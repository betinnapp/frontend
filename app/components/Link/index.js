/**
 *
 * Link
 *
 */

import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Link as RouterLink } from 'react-router-dom'

const LinkStyled = styled(RouterLink)`
  color: #767676;
  font-size: 14px;
  font-weight: 600;
  text-transform: uppercase;
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`

function Link(props) {
  return (
    <LinkStyled id={props.id} to={props.to}>
      {props.children}
    </LinkStyled>
  )
}

Link.propTypes = {
  children: PropTypes.node.isRequired,
  id: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
}

export default Link
