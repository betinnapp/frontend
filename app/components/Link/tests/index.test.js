/**
 *
 * Tests for Link
 *
 */

import React from 'react'
import { render } from 'react-testing-library'
import { BrowserRouter } from 'react-router-dom'

import Link from '../index'

describe('<Link />', () => {
  const defaultProps = {
    id: 'id123',
    to: '/test',
  }

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <BrowserRouter>
        <Link {...defaultProps}>Test</Link>
      </BrowserRouter>,
    )
    expect(firstChild).toMatchSnapshot()
  })
})
