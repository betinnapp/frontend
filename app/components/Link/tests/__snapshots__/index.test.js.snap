// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> Should render and match the snapshot 1`] = `
<a
  class="Link__LinkStyled-sc-8xsoh-0 jbHswc"
  href="/test"
  id="id123"
>
  Test
</a>
`;
